### PR TITLE
[NFC] clang-format utils/TableGen

### PR DESCRIPTION
**Pull request by Pierre-vh**
```
find llvm/utils/TableGen -iname "*.h" -o -iname "*.cpp" | xargs clang-format-16 -i
```

Split from #80847 - there is quite a bit more reformatting going on here because this is a full reformating of every file.

### CODE REVIEW

**Review by arsenm — COMMENTED** [review]
Be sure to add this to .git-blame-ignore-revs once pushed

**Comment by chapuni** [comment]
Oh sorry, I meant; "Just before committing #80847"

File: llvm/utils/TableGen/CTagsEmitter.cpp [hunk 1]
```diff
@@ -1,4 +1,5 @@
-//===- CTagsEmitter.cpp - Generate ctags-compatible index ------------------===//
+//===- CTagsEmitter.cpp - Generate ctags-compatible index
```
Comment (arsenm) [inline llvm/utils/TableGen/CTagsEmitter.cpp:1]: Broke this line

**Comment by Pierre-vh** [comment]
> Oh sorry, I meant; "Just before committing #80847"

Do you mean this should only land if #80847, @chapuni ?
I was thinking that this is a good thing nonetheless and could land even if ##80847 doesn't  - what do you think?

File: llvm/utils/TableGen/AsmMatcherEmitter.cpp [hunk 2]
```diff
@@ -1116,15 +1109,33 @@ static std::string getEnumNameForToken(StringRef Str) {
...
-    case '*': Res += "_STAR_"; break;
-    case '%': Res += "_PCT_"; break;
-    case ':': Res += "_COLON_"; break;
-    case '!': Res += "_EXCLAIM_"; break;
-    case '.': Res += "_DOT_"; break;
-    case '<': Res += "_LT_"; break;
-    case '>': Res += "_GT_"; break;
-    case '-': Res += "_MINUS_"; break;
-    case '#': Res += "_HASH_"; break;
+    case '*':
```
Comment (Pierre-vh) [inline llvm/utils/TableGen/AsmMatcherEmitter.cpp:1112]: this type of change is a bit annoying, should I revert this one and add clang-format-off/on?

File: llvm/utils/TableGen/AsmWriterEmitter.cpp [hunk 3]
```diff
@@ -239,35 +241,58 @@ FindUniqueOperandCommands(std::vector<std::string> &UniqueOperandCommands,
...
-      case 'e': Str[i] = 27; break;
-      case 'f': Str[i] = '\f'; break;
-      case 'n': Str[i] = '\n'; break;
-      case 'r': Str[i] = '\r'; break;
-      case 't': Str[i] = '\t'; break;
-      case 'v': Str[i] = '\v'; break;
-      case '"': Str[i] = '\"'; break;
-      case '\'': Str[i] = '\''; break;
-      case '\\': Str[i] = '\\'; break;
+    if (Str[i] == '\\' && i != Str.size() - 1) {
```
Comment (Pierre-vh) [inline llvm/utils/TableGen/AsmWriterEmitter.cpp:256]: idem

File: llvm/utils/TableGen/AsmWriterEmitter.cpp [hunk 4]
```diff
@@ -281,14 +306,19 @@ static void UnescapeString(std::string &Str) {
 /// causes non-standard escape character warnings.
 static void UnescapeAliasString(std::string &Str) {
   for (unsigned i = 0; i != Str.size(); ++i) {
-    if (Str[i] == '\\' && i != Str.size()-1) {
-      switch (Str[i+1]) {
-      default: continue;  // Don't execute the code after the switch.
-      case '{': Str[i] = '{'; break;
-      case '}': Str[i] = '}'; break;
+    if (Str[i] == '\\' && i != Str.size() - 1) {
```
Comment (Pierre-vh) [inline llvm/utils/TableGen/AsmWriterEmitter.cpp:309]: idem

File: llvm/utils/TableGen/AsmWriterInst.cpp [hunk 5]
```diff
@@ -57,54 +57,55 @@ AsmWriterInst::AsmWriterInst(const CodeGenInstruction &CGI, unsigned CGIIndex,
...
-          case '"':
-            AddLiteralString("\\\"");
-            break;
-          case '\\':
-            AddLiteralString("\\\\");
-            break;
-          default:
-            AddLiteralString(std::string(1, AsmString[LastEmitted]));
-            break;
+        case '\n':
```
Comment (Pierre-vh) [inline llvm/utils/TableGen/AsmWriterInst.cpp:68]: idem

File: llvm/utils/TableGen/CodeGenTarget.cpp [hunk 6]
```diff
@@ -49,10 +49,14 @@ MVT::SimpleValueType llvm::getValueType(const Record *Rec) {
 
 StringRef llvm::getName(MVT::SimpleValueType T) {
   switch (T) {
-  case MVT::Other:   return "UNKNOWN";
-  case MVT::iPTR:    return "TLI.getPointerTy()";
-  case MVT::iPTRAny: return "TLI.getPointerTy()";
-  default: return getEnumName(T);
+  case MVT::Other:
```
Comment (Pierre-vh) [inline llvm/utils/TableGen/CodeGenTarget.cpp:52]: another one

**Review by arsenm — APPROVED** [review]
lgtm assuming I got all the broken header lines. The needing to load in every file individually on GitHub is really annoying

File: llvm/utils/TableGen/InstrInfoEmitter.cpp [hunk 7]
```diff
@@ -1202,49 +1201,88 @@ void InstrInfoEmitter::emitRecord(
...
-  if (Inst.isTerminator)       OS << "|(1ULL<<MCID::Terminator)";
-  if (Inst.isReMaterializable) OS << "|(1ULL<<MCID::Rematerializable)";
-  if (Inst.isNotDuplicable)    OS << "|(1ULL<<MCID::NotDuplicable)";
-  if (Inst.Operands.hasOptionalDef) OS << "|(1ULL<<MCID::HasOptionalDef)";
-  if (Inst.usesCustomInserter) OS << "|(1ULL<<MCID::UsesCustomInserter)";
-  if (Inst.hasPostISelHook)    OS << "|(1ULL<<MCID::HasPostISelHook)";
-  if (Inst.Operands.isVariadic)OS << "|(1ULL<<MCID::Variadic)";
-  if (Inst.hasSideEffects)     OS << "|(1ULL<<MCID::UnmodeledSideEffects)";
-  if (Inst.isAsCheapAsAMove)   OS << "|(1ULL<<MCID::CheapAsAMove)";
+  if (Inst.isPreISelOpcode)
```
Comment (Pierre-vh) [inline llvm/utils/TableGen/InstrInfoEmitter.cpp:1204]: this time it's an if, might be better to also off/on clang format here?

File: llvm/utils/TableGen/X86DisassemblerTables.cpp [hunk 8]
```diff
@@ -41,8 +41,9 @@ static inline const char *stringForContext(InstructionContext insnContext) {
     break;
 #define ENUM_ENTRY_K_B(n, r, d)                                                \
   ENUM_ENTRY(n, r, d)                                                          \
-  ENUM_ENTRY(n##_K_B, r, d) ENUM_ENTRY(n##_KZ, r, d) ENUM_ENTRY(n##_K, r, d)   \
-      ENUM_ENTRY(n##_B, r, d) ENUM_ENTRY(n##_KZ_B, r, d)
+  ENUM_ENTRY(n##_K_B, r, d)                                                    \
```
Comment (Pierre-vh) [inline llvm/utils/TableGen/X86DisassemblerTables.cpp:44]: Should I just put each ENUM_ENTRY on their own line?

**Review by Pierre-vh — COMMENTED** [review]
I tried my best to tag changes I found suspicious. There's a lot of noise because some files were pretty old I assume and used different indentation rules (e.g. a class inside a namespace was indented). Removing the indentation + formatting the code itself seems to prove too much for git diffing and it shows those changes as deleted/rewritten code.

File: llvm/utils/TableGen/AsmMatcherEmitter.cpp (hunk 2 above)
Comment (arsenm) [reply llvm/utils/TableGen/AsmMatcherEmitter.cpp:1112]: No, I really hate the fold switch case onto one line pattern

**Review by chapuni — APPROVED** [review]
I supposed I had to avoid formatting them since they were really historical files.

Nowadays, we have the format checker in the CI. Fair enough for reformatting.

**Pull request merged**